### ISSUE TITLE
Is there an alternative to running evaluator on train_loader? 

### ISSUE THREAD

**Issue by karandwivedi42** [pytorch/ignite#185]
[In this line](https://github.com/pytorch/ignite/blob/master/examples/mnist_with_tensorboardx.py#L104), is there any other way to do get average train loss than to run evaluator on train_loader again. 

1. The random transforms will be different on each run so this won't give the correct exact loss.

2. It doesn't scale to something like imagenet..

Am I missing something?

**Comment by vfdev-5** [comment]
@karandwivedi42 take a look at the footnote [here](https://github.com/pytorch/ignite/blob/master/docs/source/quickstart.rst#id21) and the discussion [here](https://github.com/pytorch/ignite/pull/165):

- online train metric computation, but model is not fixed
or
- offline train metric computation on a subset of the training dataset

**Comment by karandwivedi42** [comment]
Thanks!
